greenpeace/planet4-design-tokens
# PLANET-7002: Implement CircleCI Automation of Design Tokens

Ref: https://jira.greenpeace.org/browse/PLANET-7002

This new functionality runs when new tokens are pushed to this repository. Ideally, they would be pushed directly from Figma since the new changes are supposed to be added by the P4 Design Team. 

This PR includes the full Circle CI configuration and the integration with the [planet4-master-theme repository](https://github.com/greenpeace/planet4-master-theme), which means including the new tokens into the Master Theme.

## Step by step about the integration
Whenever new (or not) tokens were pushed to this repository, then:

1. A Python script parses the `tokens.json` file into a `_tokens.scss` file. (eg. `_tokens.scss`)
2. ~[A new version is released](https://github.com/greenpeace/planet4-design-tokens/releases).~ This is no longer needed since the release will be promoted by the design team.
3. Inside the pipeline, the master theme is cloned and created a new branch with the latest pushed tag, for instance `tokens-<tag-name>`. That scss files is already included into [main style file](https://github.com/greenpeace/planet4-master-theme/blob/main/assets/src/scss/style.scss#L14).
4. Create automatically a pull request with this new changes. 
5. This will be deployed automatically on a test instance so UAT can also take place. A developer can check the PR code (the DoD?) and approve/merge the PR when UAT has passed.

### Testing
1. Commit a new change (it could be empty)
2. Go to the releases section and create a new release. You should following the `/^v.*/` pattern for tags. 
3. Go to the [CircleCi pipelines](https://app.circleci.com/pipelines/github/greenpeace/planet4-design-tokens) and check about its status.
**If the workflow `tag` is passed, then:**
1. Go to the [Master Theme repo](https://github.com/greenpeace/planet4-master-theme/pulls) and check if a new pull request has been created.
